core/src/main/scala/weaponregex/parser/Parser.scala

```diff
@@ -12,17 +12,31 @@ import scala.util.{Failure, Success, Try}
   */
 object Parser {
 
+  /** Apply the parser to parse the given pattern and flags
+    * @param pattern
+    *   The regex pattern to be parsed
+    * @param flags
+    *   The regex flags to be used
+    * @return
+    *   A `Success` of parsed [[weaponregex.model.regextree.RegexTree]] if can be parsed, a `Failure` otherwise
+    */
+  def apply(pattern: String, flags: String, flavor: ParserFlavor): Try[RegexTree] =
+    flavor match {
+      case ParserFlavorJVM =>
+        if (flags.nonEmpty)
+          println("[Warning] Parser: JVM regex flavor does not support string flags. Flags will be ignored.")
+        new ParserJVM(pattern).parse
+      case ParserFlavorJS => new ParserJS(pattern, flags).parse
+      case _              => Failure(new RuntimeException("[Error] Parser: Unsupported regex flavor"))
+    }
+
   /** Apply the parser to parse the given pattern
     * @param pattern
     *   The regex pattern to be parsed
     * @return
     *   A `Success` of parsed [[weaponregex.model.regextree.RegexTree]] if can be parsed, a `Failure` otherwise
     */
-  def apply(pattern: String, flavor: ParserFlavor = ParserFlavorJVM): Try[RegexTree] = flavor match {
-    case ParserFlavorJVM => new ParserJVM(pattern).parse
-    case ParserFlavorJS  => new ParserJS(pattern).parse
-    case _               => Failure(new RuntimeException("[Error] Parser: Unsupported regex flavor"))
-  }
+  def apply(pattern: String, flavor: ParserFlavor = ParserFlavorJVM): Try[RegexTree] = apply(pattern, "", flavor)
```

Comment:
It would signify that flags are optional. A required argument, but filled with an empty string is less clear and makes it seem like a pattern always has flags (but just empty ones), which isn't the case. Especially since it's not implemented for all parser flavours